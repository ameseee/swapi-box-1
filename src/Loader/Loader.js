import React from 'react';
import logo from '../assets/imp-logo.svg';

const Loader = () => {
  return (
    <div className="loader-div">
      <p>Loading...</p>
      <img src={logo} className="App-logo" alt="logo" />
    </div>
  );
};

export default Loader;
