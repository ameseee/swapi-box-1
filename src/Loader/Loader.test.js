import React from 'react';
import Loader from './Loader';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

describe('Loader', () => {
  Enzyme.configure({ adapter: new Adapter() });
  const wrapper = shallow(<Loader />);

  it('should exist', () => {
    const loader = wrapper.find('.loader-div');

    expect(loader.length).toEqual(1);
  });

  it('should have loading msg', () => {
    const msg = wrapper.find('p');

    expect(msg.length).toEqual(1);
    expect(msg.text()).toEqual('Loading...');
  });

  it('should have logo', () => {
    const logo = wrapper.find('img');

    expect(logo.length).toEqual(1);
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
