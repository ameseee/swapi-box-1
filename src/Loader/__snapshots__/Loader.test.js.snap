// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Loader />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p>
          Loading...
</p>,
        <img
          alt="logo"
          className="App-logo"
          src="imp-logo.svg"
/>,
      ],
      "className": "loader-div",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "Loading...",
        },
        "ref": null,
        "rendered": "Loading...",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "alt": "logo",
          "className": "App-logo",
          "src": "imp-logo.svg",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p>
            Loading...
</p>,
          <img
            alt="logo"
            className="App-logo"
            src="imp-logo.svg"
/>,
        ],
        "className": "loader-div",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "Loading...",
          },
          "ref": null,
          "rendered": "Loading...",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "alt": "logo",
            "className": "App-logo",
            "src": "imp-logo.svg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
